assets/src/edit-story/components/carousel/useCarouselScroll.js

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useCallback, useState, useLayoutEffect } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useConfig } from '../../app';
+
+function useCarouselScroll({ listElement, carouselWidth, hasOverflow }) {
+  const [scrollPercentage, setScrollPercentage] = useState(0);
```

Comment:
Nit: this value doesn't seem to be in percentage actually. Not sure what would be a better name either so unless you have a more accurate one, can probably stay as is.

Reply:
`scrollRatio`?

Reply:
Fixed in b1d4e1472.